rust/ffi/node/src/error.rs

```diff
@@ -22,8 +22,15 @@ use snafu::Snafu;
 pub enum Error {
     #[snafu(display("column '{name}' is missing"))]
     MissingColumn { name: String },
+    #[snafu(display("{name}: {message}"))]
+    RangeError { name: String, message: String },
```

Comment:
Should range error convert to a Javascript RangeError? https://docs.rs/neon/latest/neon/context/trait.Context.html#method.throw_range_error

Reply:
I see. That's because `ResultExt` is generic over `E`, so you can't match on that specific variant.

I guess the only thing you could do is write a special `or_throw` method just for `crate::Error`. But it might not be worth it; the error types in JS seems so limited I doubt people are really caring about the type.